### PR TITLE
#101 Add context metadata

### PR DESCRIPTION
**Pull request by kaklakariada**
Part of #101

### CODE REVIEW

File: extension-manager-client-java/src/test/java/com/exasol/extensionmanager/client/model/EqualsContractTest.java [hunk 1]
```diff
@@ -1,17 +1,49 @@
...
+
+    private static Stream<Class<?>> findModelClasses() throws IOException {
+        final String modelPath = MODEL_PACKAGE.replace(".", "/");
+        final Path sourcePath = GENERATED_SOURCES.toAbsolutePath().resolve(modelPath);
+        return Files.list(sourcePath).map(Path::getFileName).map(Path::toString) //
+                .filter(fileName -> fileName.endsWith(".java"))
+                .map(fileName -> fileName.substring(0, fileName.lastIndexOf(".")))
+                .map(className -> MODEL_PACKAGE + "." + className)
+                .map(qualifiedClassName -> loadClass(qualifiedClassName));
+    }
```
Comment (kaklakariada) [inline extension-manager-client-java/src/test/java/com/exasol/extensionmanager/client/model/EqualsContractTest.java:40]: Ensure that we always test all model classes even if new ones are added (=generated)

File: pkg/extensionAPI/context/bucketfs.go
Comment (kaklakariada) [inline pkg/extensionAPI/context/bucketfs.go:1]: Moved from `context.go` to separate file in same package

File: pkg/extensionAPI/context/sqlClient.go
Comment (kaklakariada) [inline pkg/extensionAPI/context/sqlClient.go:1]: Moved to separate file from `context.go`

File: pkg/extensionAPI/exaMetadata/ExaMetadata.go [hunk 2]
```diff
@@ -50,22 +77,30 @@ WHERE SCRIPT_SCHEMA=?`, schemaName)
...
 	}
 	return &ExaScriptTable{Rows: rows}, nil
 }
 
+func readScriptRow(result *sql.Rows) (*ExaScriptRow, error) {
+	var row ExaScriptRow
+	var inputType sql.NullString
+	var resultType sql.NullString
+	var comment sql.NullString
+	err := result.Scan(&row.Schema, &row.Name, &row.Type, &inputType, &resultType, &row.Text, &comment)
```
Comment (kaklakariada) [inline pkg/extensionAPI/exaMetadata/ExaMetadata.go:94]: `sql.NullString` is required for `Nullable` columns

File: pkg/extensionController/controller_test.go [hunk 3]
```diff
@@ -67,6 +67,8 @@ func (suite *ControllerUTestSuite) initDbMock() {
 
 func (suite *ControllerUTestSuite) AfterTest(suiteName, testName string) {
 	suite.NoError(suite.dbMock.ExpectationsWereMet())
+	suite.bucketFsMock.AssertExpectations(suite.T())
+	suite.metaDataMock.AssertExpectations(suite.T())
```
Comment (kaklakariada) [inline pkg/extensionController/controller_test.go:71]: Was missing. This did hide invalid mocking which is fixed below.

File: pkg/extensionController/controller_i_test.go [hunk 4]
```diff
@@ -160,6 +160,15 @@ func (suite *ControllerITestSuite) TestUpgradeSucceeds() {
...
 }
 
+func (suite *ControllerITestSuite) TestUpgradeGettingScriptReturnsNil() {
+	suite.createExtensionBuilder().
+		WithUpgradeFunc("const script = context.metadata.getScriptByName('script'); return {previousVersion:'0.1.0',newVersion:'script is null = '+(script===null)};").
+		Build().WriteToFile(path.Join(suite.tempExtensionRepo, EXTENSION_ID))
+	result, err := suite.createController().UpgradeExtension(mockContext(), suite.exasol.GetConnection(), EXTENSION_ID)
+	suite.NoError(err)
+	suite.Equal(&extensionAPI.JsUpgradeResult{PreviousVersion: "0.1.0", NewVersion: "script is null = true"}, result)
+}
```
Comment (ckunki) [inline pkg/extensionController/controller_i_test.go:170]: Sorry, my suggestion is probably not working, but I hope it expresses my proposal ? 

```suggestion
func (suite *ControllerITestSuite) TestUpgradeGettingScriptReturnsNil() {
    String UNINSTALLED_SCRIPT = script;
    script_not_installed = "(script===null)";
	suite.createExtensionBuilder().
	    WithUpgradeFunc("const installation = context.metadata.getScriptByName('" //
			    +UNINSTALLED_SCRIPT //
			    'result = installation === null ? "null" : "other"; '
			    + "'); return {previousVersion:'0.1.0',newVersion:'result};").
		Build().WriteToFile(path.Join(suite.tempExtensionRepo, EXTENSION_ID))
	result, err := suite.createController().UpgradeExtension(mockContext(), suite.exasol.GetConnection(), EXTENSION_ID)
	suite.NoError(err)
	suite.Equal(&extensionAPI.JsUpgradeResult{PreviousVersion: "0.1.0", NewVersion: "script is null = true"}, result)
}
```

**Review by ckunki — CHANGES_REQUESTED** [review]

**Review by ckunki — APPROVED** [review]

**Pull request merged**